instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex

```diff
@@ -162,21 +162,23 @@ defmodule OpentelemetryPhoenix do
         %{kind: kind, reason: reason, stacktrace: stacktrace} = meta,
         _config
       ) do
-    ctx = OpentelemetryTelemetry.set_current_telemetry_span(@tracer_id, meta)
+    if OpenTelemetry.Span.is_recording(OpenTelemetry.Tracer.current_span_ctx()) do
+      ctx = OpentelemetryTelemetry.set_current_telemetry_span(@tracer_id, meta)
 
-    {[reason: reason], attrs} =
-      Reason.normalize(reason)
-      |> Keyword.split([:reason])
+      {[reason: reason], attrs} =
+        Reason.normalize(reason)
+        |> Keyword.split([:reason])
 
-    # try to normalize all errors to Elixir exceptions
-    exception = Exception.normalize(kind, reason, stacktrace)
+      # try to normalize all errors to Elixir exceptions
+      exception = Exception.normalize(kind, reason, stacktrace)
 
-    # record exception and mark the span as errored
-    Span.record_exception(ctx, exception, stacktrace, attrs)
-    Span.set_status(ctx, OpenTelemetry.status(:error, ""))
+      # record exception and mark the span as errored
+      Span.record_exception(ctx, exception, stacktrace, attrs)
+      Span.set_status(ctx, OpenTelemetry.status(:error, ""))
```

Comment:
Looks like there's some kind of whitespace thing going on here?

Reply:
There is a big `if` statement around the whole thing, that's where the extra indenting is coming from.